src/test/regress/expected/intermediate_result_pruning.out

```diff
@@ -830,7 +831,7 @@ INSERT INTO table_1
 	 ) foo
 	 where table_2.key != 1 AND
  	foo.key = table_2.value::int;
-DEBUG:  Set operations are not allowed in distributed INSERT ... SELECT queries
```

Comment:
Do we have any tests like:

INSERT INTO TEST SELECT * FROM test UNION SELECT * FROM test;

I'm trying to understand why we disallowed SET operation pushdown in the past.  It might be some deparsing problems.

Try also:

INSERT INTO TEST b,a SELECT b,a FROM test UNION ALL SELECT b,a FROM test;

Normally these should be fine to pushdown, but apperatenly do not on main

Reply:
I think, ideally, we can push down the first query but `ContainsUnionSubquery` returns false since the union query does not contain FROM clause. Then `RestrictionEquivalenceForPartitionKeys` check throws error as expected. INSERT SELECT is handled by repartition as fallback though.

But still, we should let union operations be pushed down if possible for INSERT SELECT. When not possible, they will be pulled to coordinator or repartitioned.